dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResource.java

```diff
@@ -1004,4 +1009,57 @@ public Response getPageTypes(@Context final HttpServletRequest originalRequest,
         return paginationUtil.getPage(originalRequest, user, filter, page, perPage, orderbyParam,
                     OrderDirection.valueOf(direction), extraParams);
     }
+
+    /**
+     * Returns true if the page exist and the current user has 'type' permission over it
+     * Parameters:
+     * - type: is optional, by default is READ
+     * - path: page path
+     *
+     * @param originalRequest The {@link HttpServletRequest} object.
+     * @param response The {@link HttpServletResponse} object.
+     * @param type {@link String} type: READ by default @see {@link PermissionLevel}
+     * @param path {@link String} page path
+     * @return All the content types that match
+     */
+    @NoCache
+    @POST
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    @Path("/_check-permission")
+    public ResponseEntityBooleanView checkPagePermission(@Context final HttpServletRequest request,
+                                                         @Context final HttpServletResponse response,
+                                                         final PageCheckPermissionForm pageCheckPermissionForm) throws DotSecurityException, DotDataException {
+
+        final User user = webResource.init(request, response, true).getUser();
```

Comment:
Always use the InitBuilder rather than the `webResource.init(xxx)` as they make the permissions required explicit.  We need to deprecate all the `webResource.init` methods.